### PR TITLE
Add a yamllint presubmit to test-infra

### PR DESCRIPTION
**Pull request by alvaroaleman**
This PR adds a yamllint presubmit to test-infra and fixes up the issues yamllint found.

I used a Docker images from ours (quay.io/kubermatic/yamllint:0.1) because I don't know the process of adding a new docker-image for test-infra usage. Its Dockerfile looks like this:

```Dockerfile
FROM python:3.7.1-alpine3.8

RUN pip3 install yamllint==1.13.0
```

If someone tells me how image publishing works I am happy to add that in this PR too.

Some of the default linters that I didn't disable can be argued about, specifically:
* trailing spaces
* trailing blank lines
* no new line character at the end of file

But since the amount of issues that had to be fixed because of them was still limited and it increases readability I left them enabled. No strong opinion here thought.

The linters that are enabled now are:

```
rules:
  braces: enable
  brackets: enable
  colons: enable
  commas: enable
  comments-indentation:
    level: warning
  empty-lines: enable
  empty-values: enable
  hyphens: enable
  key-duplicates: enable
  new-line-at-end-of-file: enable
  new-lines: enable
  octal-values: enable
  quoted-strings: disable
  trailing-spaces: enable
  truthy:
    level: warning
```

More details about the meaning of the rules can be found [in the yamllint docs](https://yamllint.readthedocs.io/en/stable/rules.html)

/assign @fejta

### CODE REVIEW

File: config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml [hunk 1]
```diff
@@ -229,3 +229,15 @@ presubmits:
       - image: gcr.io/k8s-testimages/kubekins-e2e:v20190420-93fab49-experimental
         command:
         - ./hack/verify-github-spelling.sh
+
+  - name: pull-test-infra-yamllint
```
Comment (fejta) [inline config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml]: This will likely need an entry in testgrid/config.yaml

**Comment by alvaroaleman** [comment]
Added the testgrid config.

Thinking about this again, we could actually have it run for the whole repo and not just `config/jobs`. Right now that yields a lot more errors that would have to be fixed thought. WDYT?

File: config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml [hunk 2]
```diff
@@ -229,3 +229,15 @@ presubmits:
...
+    always_run: true
+    decorate: true
+    spec:
+      containers:
+      - image: quay.io/kubermatic/yamllint:0.1
+        command:
+        - yamllint
```
Comment (fejta) [inline config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml]: Is there an automatic delinter?

Typically we have a hack/update-yaml.sh and hack/verify-yaml.sh scripts to update and validate this sort of thing.

Comment (alvaroaleman) [reply config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml]: Not to my knowledge. There are also some rules for which this wouldn't really be possible, e.G. `duplicate keys`

**Comment by fejta** [comment]
Generally LGTM, let's see if there's any feedback from other test-infra contributors

**Comment by Katharine** [comment]
Seems like a good idea to me.

**Comment by spiffxp** [comment]
/lgtm
I'll leave the /approve to someone else

I don't think this makes things worse, and I enjoy automating away style wars, so +1 to that.  

But I don't think it covers the gap we hit with an incorrect prowjob config.  Either we didn't bother unmarshalling the prowjob yaml in a presubmit, or our unmarshalling code is tolerant of dupe/incorrect keys.  I feel like one or both of those things should change?

**Comment by stevekuznetsov** [comment]
The kubernetes apiserver is tolerant of duplicate keys -- so we have no basis to reject that PR by marshaling alone

**Comment by alvaroaleman** [comment]
> But I don't think it covers the gap we hit with an incorrect prowjob config. Either we didn't bother unmarshalling the prowjob yaml in a presubmit, or our unmarshalling code is tolerant of dupe/incorrect keys. I feel like one or both of those things should change?

The "real" solution to this problem is to use `yaml.UnmarshalStrict` for prows config which I would btw love to see. Ben did a PR for that which didn't get merged, althought it looks like that only happened due to lack of capacity on his side to finish it: https://github.com/kubernetes/test-infra/pull/10812

**Comment by alvaroaleman** [comment]
Should we merge this? Extend the PR to yamllint all of this repo? Change the PR to not use the `quay.io/kubermatic/yamllint:0.1` but one that is under control of test-infra?

**Comment by stevekuznetsov** [comment]
I am happy with this as-is, we can do the image change as a follow-up. the strict YAML parsing is hard to swallow as it means you can't roll out new features, we need versioned config

/hold cancel

**Pull request merged**